swim.go

```diff
@@ -65,7 +67,11 @@ func (n *Node) runTick() {
 			pingTimer.Reset(pingTimeout)
 			n.send(n.tick())
 		case <-pingTimer.C:
+			n.mu.Lock()
 			n.send(n.fsm.timeout())
+			n.mu.Unlock()
```

Comment:
Why do you have lock/unlock calls around this call to send but not the one above? Maybe they should be moved inside the send method?

On looking closer, it seems like send calls getAddrs which tries to acquire the mutex again? I would expect this to deadlock.

Might be worth doing an inventory of all the places in the code that touch the mutable state, and seeing if you can come up with some light conventions about how to organize the code so it's easier to know where/when to acquire the mutex.